File: src/main/java/com/intetics/lukyanenko/web/OrderController.java
package com.intetics.lukyanenko.web;

import com.intetics.lukyanenko.models.Customer;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.servlet.ModelAndView;

import javax.inject.Inject;
import java.util.HashMap;
import java.util.Map;

@Controller
@RequestMapping(value = "/order")
public class OrderController
{
  private final AppService service;
  
  @Inject
  public OrderController(AppService service)
  {
    this.service = service;
  }
  
  @RequestMapping(value = "/{id}", method = RequestMethod.GET)
  public ModelAndView get(@RequestParam Integer id)
  {
    return new ModelAndView("OrderEdit", "model", service.getOrder(id));
  }
  
  @RequestMapping(value = "/{id}/cancel", method = RequestMethod.POST)
  public String cancel(@PathVariable("id") Integer id)
  {
    service.deleteOrder(id);
    return "redirect:/order";
  }
  
  @RequestMapping(value = {"/{id}", ""}, method = RequestMethod.POST)
  public ModelAndView update(@PathVariable("id") Integer orderId,
                             @RequestParam       Map<String, String> params)
  {
    service.updateOrder(orderId, params);
    return new ModelAndView("OrderEdit", "model", service.getOrder(orderId));
  }
  
  @RequestMapping(value = {"/", ""}, method = RequestMethod.GET)
  public ModelAndView getList()
  {
    Customer customer = null; // todo Get customer from security context
    return new ModelAndView("OrderList", "model", service.getOrderList(customer));
  }
  
  @RequestMapping(value = "/{orderId}/deleteItem/{itemId}")
  public void deleteItem(@PathVariable("orderId") Integer orderId,
                         @PathVariable("ItemId")  Integer itemId)
  {
    HashMap<String, String> params = new HashMap<>(1);
    params.put(String.valueOf(itemId), "0");
    service.updateOrder(orderId, params);
  }
  
}

File: src/main/java/com/intetics/lukyanenko/web/OrderController.kt
package com.intetics.lukyanenko.web

import com.intetics.lukyanenko.models.Customer
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.servlet.ModelAndView

import javax.inject.Inject

@Controller
@RequestMapping(value = "/order")
class OrderController @Inject
constructor(private val service: AppService) {

    @RequestMapping(value = "/{id}", method = arrayOf(RequestMethod.GET))
    fun get(@RequestParam id: Int): ModelAndView {
        return ModelAndView("OrderEdit", "model", service.getOrder(id))
    }

    @RequestMapping(value = "/{id}/cancel", method = arrayOf(RequestMethod.POST))
    fun cancel(@PathVariable("id") id: Int): String {
        service.deleteOrder(id)
        return "redirect:/order"
    }

    @RequestMapping(value = *arrayOf("/{id}", ""), method = arrayOf(RequestMethod.POST))
    fun update(@PathVariable("id") orderId: Int,
               @RequestParam params: Map<String, String>): ModelAndView {
        service.updateOrder(orderId, params)
        return ModelAndView("OrderEdit", "model", service.getOrder(orderId))
    }

    // todo Get customer from security context
    @RequestMapping(value = *arrayOf("/", ""), method = arrayOf(RequestMethod.GET))
    fun list(): ModelAndView {
        return ModelAndView("OrderList" /*, "model", service.getOrderList(customer)*/)
    }

    @RequestMapping(value = "/{orderId}/deleteItem/{itemId}")
    fun deleteItem(@PathVariable("orderId") orderId: Int,
                   @PathVariable("ItemId") itemId: Int) {
        service.updateOrder(orderId, mapOf(itemId.toString() to "0"))
    }
}
